File: corpusforge-frontend/src/types/query.ts
export type Confidence = 'High' | 'Medium' | 'Low';

export interface Citation {
  document_id: string;
  filename: string;
  // Optional so the Heat Stamp (CitationChip) can also be used for graph-connection sources,
  // which carry a source document but no page number.
  page_number?: number;
}

export interface QueryResponse {
  answer: string;
  confidence: Confidence;
  citations: Citation[];
  used_graph: boolean;
  follow_ups: string[];
}

export interface ChatMessage {
  id: string;
  type: 'question' | 'answer';
  content: string;
  response?: QueryResponse;
  timestamp: Date;
}

File: corpusforge-backend/schemas/query.py
from pydantic import BaseModel, field_validator


class QueryRequest(BaseModel):
    question: str

    @field_validator("question")
    @classmethod
    def validate_question(cls, v: str) -> str:
        v = v.strip()
        if not v:
            raise ValueError("Question cannot be empty.")
        if len(v) > 500:
            raise ValueError("Question must be 500 characters or fewer.")
        return v


class Citation(BaseModel):
    document_id: str
    filename: str
    page_number: int


class QueryResponse(BaseModel):
    answer: str
    confidence: str
    citations: list[Citation]
    used_graph: bool
    follow_ups: list[str]
